internal/graph/cycle_check_resolver.go

```diff
@@ -59,6 +59,7 @@ func (c *CycleDetectionCheckResolver) ResolveCheck(
 		RequestMetadata:      req.GetRequestMetadata(),
 		VisitedPaths:         req.VisitedPaths,
 		Context:              req.GetContext(),
+		Consistency:          req.GetConsistency(),
```

Comment:
Instead of making this change you could change to `return c.delegate.ResolveCheck(ctx, clone(req))`

Reply:
Done in c3c6d511e255f424d3c85e7a61f9e376cbc5f478